docs/source/links/rest.md

```diff
@@ -0,0 +1,148 @@
+---
+title: apollo-link-rest
+description: Call your REST APIs inside your graphQL queries.
+---
+
+[**Read the announcement post! ( to be pusblished soon )
+🎉**](https://dev-blog.apollodata.com/)
+
+If you have a REST API, you can now query your endpoints using GraphQL and have all your data managed by Apollo Client. There is no need to write a wrapper around your REST API.
+
+<h2 id="start">Quick start</h2>
+
+To get started, install `apollo-link-rest` from npm:
+
+```bash
+npm install apollo-link-rest --save
+```
+
+The rest of the instructions assume that you have already [set up Apollo
+Client](/docs/react/basics/setup.html#installation) in your application. After
+you install the package, you can create your rest link.
+
+```js
+const link = new RestLink({ uri: "https://api.example.com" });
+```
+
+<h2 id="options">Options</h2>
+
+REST Link takes an object with some options on it to customize the behavior of the link. The options you can pass are outlined below:
+
+- `uri`: the URI key is a string endpoint (optional when `endpoints` provides a default)
+- `endpoints`: root endpoint (uri) to apply paths to or a map of endpoints
+- `customFetch`: a custom `fetch` to handle REST calls
+- `headers`: an object representing values to be sent as headers on the request
+- `credentials`: a string representing the credentials policy you want for the fetch call
+- `fieldNameNormalizer`: function that takes the response field name and converts it into a GraphQL compliant name
+
+<h2 id="context">Context</h2>
+
+REST Link uses the `headers` field on the context to allow passing headers to the HTTP request. It also supports the `credentials` field for defining credentials policy.
+
+- `headers`: an object representing values to be sent as headers on the request
+- `credentials`: a string representing the credentials policy you want for the fetch call
+
+<h2 id="query">Writing your first query</h2>
+
+To write your first query, add a `@rest` directive to the field that should be resolved from your rest endpoint:
+
+```js
+const query = gql`
+  query example {
+    repos @rest(type: "Repo", path: "/repos") {
+      id
+      name
+    }
+  }
+`;
+```
+
+<h2 id="rest">@rest directive</h2>
+
+This is where you setup the endpoint you want to fetch.
+The rest directive could be used at any depth in a query, but once it is used, nothing nested in it can be GraphQL data, it has to be from the rest link or other resource (like a @client directive)
```

Comment:
This is unclear -- what do you mean by `nothing nested in it can be GraphQL data`?

Reply:
I want to emphasize that all subfields inside Rest query will be swallow by the directive.

If writing for example 

```js
query MyQuery {
  planets @rest(type: "PlanetPayload", path: "planets/") {
    count
    next
    results {
      name
    }
  }
}
```

you can't have `results` to be resolved in an other link whereas the contrary is true

Reply:
Alternate wording proposal:

>The `@rest`-directive can be mixed in at any depth in a complex query, however one current limitation is that you can't chain any other links to resolve nested fields.
> Example:
```graphql
query MyQuery {
  planets @rest(type: "PlanetPayload", path: "planets/") {
    count
    next
    results @SomeNonRestDirective_or_alternate_ApolloLink_Is_Unsupported {
      name
    }
  }
}
```